pkg/controller/namespace/deletion/namespaced_resources_deleter.go

```diff
@@ -163,7 +163,7 @@ func (d *namespacedResourcesDeleter) initOpCache(clusterName logicalcluster.Name
 		utilruntime.HandleError(fmt.Errorf("unable to get all supported resources from server: %v", err))
 	}
 	if len(resources) == 0 {
-		klog.Fatalf("Unable to get any supported resources from server: %v", err)
+		return fmt.Errorf("unable to get any supported resources from server: %v", err)
```

Comment:
+1, previously it could have killed the server.

Reply:
it did in my test.